src/plugins/home/server/services/sample_data/routes/install.ts

```diff
@@ -198,14 +200,18 @@ export function createInstallRoute(
       }
 
       let createResults;
-      const savedObjectsList = dataSourceId
-        ? sampleDataset.getDataSourceIntegratedSavedObjects(dataSourceId, dataSourceTitle)
-        : sampleDataset.savedObjects;
+      const savedObjectsList =
+        dataSourceId || workspaceId
+          ? sampleDataset.getWorkspaceAndDataSourceIntegratedSavedObjects(workspaceId)(
+              dataSourceId,
+              dataSourceTitle
+            )
+          : sampleDataset.savedObjects;
```

Comment:
Can we make it simply function call here?

```typescript
let savedObjectsList = sampleDataset.savedObjects;
if (workspaceId) {
  savedObjectsList = getSavedObjectsWithWorkspaceId(savedObjectsList, workspaceId)
}

if (dataSourceId) {
  savedObjectsList = getSavedObjectsWithDataSource(savedObjectsList, dataSourceId, dataSourceTitle)
}
```
I think this will simplify the logic.